sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -290,3 +291,63 @@ object HourExpressionBuilder extends ExpressionBuilder {
     }
   }
 }
+
+case class CurrentTime(precision: Int = 6) extends CurrentTimestampLike with ExpectsInputTypes {
+  // The function returns a TIME(n).
+  override def dataType: DataType = TimeType(precision)
+
+  override def prettyName: String = "current_time"
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(IntegerType)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (precision < TimeType.MIN_PRECISION || precision > TimeType.MICROS_PRECISION) {
+      TypeCheckFailure(s"Invalid precision $precision. Must be between 0 and 6.")
+    } else {
+      TypeCheckSuccess
+    }
+  }
+}
+
+/**
+ * Returns the current time at the start of query evaluation.
+ * There is no code generation since this expression should get constant folded by the optimizer.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_() - Returns the current time at the start of query evaluation. All calls of current_time within the same query return the same value.
+
+    _FUNC_ - Returns the current time at the start of query evaluation.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_();
+       15:49:11.914120
+      > SELECT _FUNC_;
+       15:49:11.914120
+      > SELECT _FUNC_(0);
+       15:49:11
+      > SELECT _FUNC_(3);
+       15:49:11.914
+  """,
+  group = "datetime_funcs",
+  since = "4.1.0")
+// scalastyle:on line.size.limit
+object CurrentTimeExpressionBuilder extends ExpressionBuilder {
```

Comment:
You can just add constructors with different parameters to `CurrentTime` instead of using `ExpressionBuilder`.

Reply:
Makes sense! Rev 2 accordingly.